File: hackerrank/30_day_of_code/day_7_arrays.js
// https://www.hackerrank.com/challenges/30-arrays/problem
"use strict";

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", inputStdin => {
    inputString += inputStdin;
});

process.stdin.on("end", _ => {
    inputString = inputString
        .replace(/\s*$/, "")
        .split("\n")
        .map(str => str.replace(/\s*$/, ""));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

const solution = a => a.reverse().join(" ");

function main() {
    const n = parseInt(readLine(), 10);

    const arr = readLine()
        .split(" ")
        .map(arrTemp => parseInt(arrTemp, 10));
    console.log(solution(arr));
}

File: hackerrank/30_day_of_code/day_7_arrays.py
# https://www.hackerrank.com/challenges/30-arrays/problem
#!/bin/python3

import math
import os
import random
import re
import sys


def solution(a):
    return ' '.join(map(str, reversed(a)))


if __name__ == '__main__':
    n = int(input())

    arr = list(map(int, input().rstrip().split()))
    print(solution(arr))
